generateSettings.ts

```diff
@@ -215,4 +232,26 @@ text += ';\n';
 text += `export const settingsSchema = ${JSON.stringify(createSchema(true))};`;
 writeFileSync(`lib/settingsGenerated.ts`, text);
 
-writeFileSync(`settings.schema.json`, JSON.stringify(createSchema(false)));
\ No newline at end of file
+writeFileSync(`settings.schema.json`, JSON.stringify(createSchema(false)));
+
+let readmeText = readFileSync('README-input.md', {encoding: 'utf-8'}) + '\n';
+function ISchemaObject(schema: Schema): schema is ISchemaObject {
+  return schema.type === 'object' && Object.values(schema.properties).length > 0;
+}
+function schemaWalk(schema: Schema, path: string[]) {
+  if (ISchemaObject(schema)) {
+    if (path.length > 0) {
+      readmeText += '#'.repeat(path.length + 1) + ` ${path.join('.')}\n`;
+      readmeText += `|name|description|type|default|\n`;
+      readmeText += `|---|---|---|---|\n`;
+    }
+    for (const [key, value] of Object.entries(schema.properties)) {
+      schemaWalk(value, [...path, key]);
+    }
+  } else {
+    readmeText += `|${path.join('.')}|${schema.description}|${schema.type}|${JSON.stringify(schema.default) ?? ''}\n`;
+  }
+}
+schemaWalk(createSchema(false), []);
+
+writeFileSync('README.md', readmeText);
```

Comment:
The toplevel `semanticTokens` and `semanticTokensDirectionColoring` somehow are added to the `analysis` section.